op-batcher/batcher/driver.go

```diff
@@ -867,9 +873,34 @@ func (l *BatchSubmitter) sendTx(txdata txData, isCancel bool, candidate *txmgr.T
 		candidate.GasLimit = intrinsicGas
 	}
 
+	floorDataGas, err := floorDataGas(candidate.TxData)
+	if err == nil && floorDataGas > candidate.GasLimit {
+		candidate.GasLimit = floorDataGas
+	}
```

Comment:
This works as a quick fix. We could add more logging to not silently ignore errors, like I just did in https://github.com/ethereum-optimism/optimism/commit/6d31e62e9ee6490da6e4548f4993ebf5149a750b

Reply:
[9c05eca](https://github.com/ethereum-optimism/optimism/pull/14517/commits/9c05eca088d4f39029ca2e0650a198e190c54ca4)